synapse/http/client.py

```diff
@@ -819,6 +820,128 @@ def __init__(
             )
 
 
+class SimpleReplicationClient(BaseHttpClient):
+    """No frills client for connecting to Replication endpoints.
+
+    Uses existing BaseHttpClient methods but replaces the 'agent' used to make the
+    request with one that supports HTTP and HTTPS.
+    Attributes:
+            endpoints.
+        agent: The custom Twisted Agent used for constructing the connection.
+    """
+
+    def __init__(
+        self,
+        hs: "HomeServer",
+    ):
+        """
+        Args:
+            hs
+        """
+        super().__init__(hs)
+
+        # Use a pool, but a very small one.
+        pool = HTTPConnectionPool(self.reactor)
+        pool.maxPersistentPerHost = 5
+        pool.cachedConnectionTimeout = 2 * 60
+
+        self.agent: IAgent = ReplicationAgent(
+            hs.get_reactor(),
+            hs,
+            contextFactory=hs.get_http_client_context_factory(),
+            pool=pool,
+        )
+
+    async def request(
+        self,
+        method: str,
+        uri: str,
+        data: Optional[bytes] = None,
+        headers: Optional[Headers] = None,
+    ) -> IResponse:
+        """
+        Args:
+            method: HTTP method to use.
+            uri: URI to query.
+            data: Data to send in the request body, if applicable.
+            headers: Request headers.
+
+        Returns:
+            Response object, once the headers have been read.
+
+        Raises:
+            RequestTimedOutError if the request times out before the headers are read
+
+        """
+        outgoing_requests_counter.labels(method).inc()
+
+        # log request but strip `access_token` (AS requests for example include this)
+        logger.debug("Sending request %s %s", method, redact_uri(uri))
+
```

Comment:
Is it necessary to `redact_uri()` a HTTP replication request? I did not think access_token was encoded into the uri outside of AS requests. Perhaps this should stay

Reply:
I don't think so.

Reply:
Removed from three places in [fedd7db](https://github.com/matrix-org/synapse/pull/15470/commits/fedd7dbdb49d30be33132b481e7a6323ad70e643)